span.go

```diff
@@ -445,12 +451,12 @@ func (s *Span) applySamplingDecision(decision SamplingDecision, lock bool) {
 
 // Span can be written to if it is sampled or the sampling decision has not been finalized.
 func (s *Span) isWriteable() bool {
-	state := s.context.samplingState
+	state := s.SpanContext().samplingState
```

Comment:
`isWriteable` and  `isSamplingFinalized` are called beneath `setTagInternal` which receives a `lock` param that is ignored here.